src/modules/personenkontext/specification/nur-rolle-lern.ts

```diff
@@ -0,0 +1,38 @@
+import { RolleRepo } from '../../rolle/repo/rolle.repo.js';
+import { DBiamPersonenkontextRepo } from '../persistence/dbiam-personenkontext.repo.js';
+import { RollenArt } from '../../rolle/domain/rolle.enums.js';
+import { Personenkontext } from '../domain/personenkontext.js';
+import { Rolle } from '../../rolle/domain/rolle.js';
+
+export class CheckRollenartLernSpecification {
+    public constructor(
+        private readonly personenkontextRepo: DBiamPersonenkontextRepo,
+        private readonly rolleRepo: RolleRepo,
+    ) {}
+
+    public async checkRollenartLern(personenkontext: Personenkontext<false>): Promise<boolean> {
+        // Fetch all personenkontexte for the person
+        const existingPersonenkontexte: Personenkontext<true>[] = await this.personenkontextRepo.findByPerson(
+            personenkontext.personId,
+        );
+
+        // Check if any existing Personenkontext has a role of type LERN
+        const rollen: Option<Rolle<true>>[] = await Promise.all(
+            existingPersonenkontexte.map((pk: Personenkontext<true>) => this.rolleRepo.findById(pk.rolleId)),
+        );
+
+        const hasLernRolle: boolean = rollen.some((rolle: Option<Rolle<true>>) => rolle?.rollenart === RollenArt.LERN);
+
+        if (hasLernRolle) {
+            // If any existing Personenkontext has a role of type LERN,
+            // check if the new Personenkontext also has a role of type LERN
+            const newRolle: Option<Rolle<true>> = await this.rolleRepo.findById(personenkontext.rolleId);
+
+            if (newRolle && newRolle.rollenart !== RollenArt.LERN) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
Maybe I'm missing something, but what about the following case:
User has multiple Personenkontexte, none of them with a Rolle of type LERN.
But the newly added Personenkontext has Rollentyp LERN. This should not be allowed, right?

I don't think this specification handles that case, is it handled somewhere else?

Reply:
You're right, this specific detail came in later as a new requirement and it was only added to the update aggregate.

Reply:
Alright, so that needs to be changed. Which is also exactly the reason why I don't like having the logic in two places 😅

Reply:
Done! logic is now restricted to one place for both endpoints.